### PR TITLE
Throttle fetch requests in the setting registry's data connector

### PR DESCRIPTION
**Pull request by ikiw**
Refactor PR #7879 to use throttler

## References
#7879 

## Code changes
Refactor of older PR #7879, to make use of Throttler in this use case as it solves the purpose of executing the callback only once even though invoked multiple times and making the code more cleaner.

Code change depends on jupyterlab/lumino#54 @afshin 

## Backwards-incompatible changes
NO

### CODE REVIEW

**Comment by blink1073** [comment]
Dependencies updated.

@ikiw, you'll need to run the  following to update dependencies:

```bash
jlpm run update:dependency @lumino/polling ^latest
jlpm run update:dependency @lumino/datagrid ^latest
```

**Comment by afshin** [comment]
@ikiw I rebased against `master`, updated the Lumino dependencies, and moved the throttling into the setting registry's data connector like we spoke about a few weeks ago on the weekly call. Sorry it took so long!

**Comment by afshin** [comment]
Docs failure is probably related to: https://github.com/minrk/pytest-check-links/pull/7

File: packages/apputils-extension/src/settingconnector.ts [hunk 1]
```diff
@@ -19,8 +21,24 @@ export class SettingConnector extends DataConnector<
...
   fetch(id: string): Promise<ISettingRegistry.IPlugin | undefined> {
-    return this._connector.fetch(id);
+    const throttlers = this._throttlers;
+    if (!(id in throttlers)) {
+      throttlers[id] = new Throttler(async () => {
+        const fetched = await this._connector.fetch(id);
+        throttlers[id].dispose();
+        delete throttlers[id];
+        return fetched;
+      }, 100);
```
Comment (saulshanabrook) [inline packages/apputils-extension/src/settingconnector.ts]:
```suggestion
      throttlers[id] = new Throttler(() => this._connector.fetch(id), 100);
```

**Comment by saulshanabrook** [comment]
Tested this in Binder, looks like it properly throttles!

Screen Shot 2020-03-25 at 1 49 02 PM

**Comment by saulshanabrook** [comment]
@meeseeksdev backport to 2.0.x

**Pull request merged**